mllib/src/main/scala/org/apache/spark/ml/feature/StopWordsRemover.scala

```diff
@@ -95,8 +121,7 @@ class StopWordsRemover @Since("1.5.0") (@Since("1.5.0") override val uid: String
         terms.filter(s => !stopWordsSet.contains(s))
       }
     } else {
-      // TODO: support user locale (SPARK-15064)
-      val toLower = (s: String) => if (s != null) s.toLowerCase else s
+      val toLower = (s: String) => if (s != null) s.toLowerCase(new Locale($(locale))) else s
```

Comment:
Maybe move `new Locale($(locale))` out of `toLower`, so we don't need to create it every time.